### ISSUE TITLE
USER ISSUE: Taking a cart down a slope sometimes gets you airborne

### ISSUE THREAD

**Issue by Falor** [StrangeLoopGames/EcoIssues#8871]
**Version:** 0.7.6.3 beta

**Steps to Reproduce:**
Place a number of ramps (no exact number as the bug occures in several locations) The material below was mainly Stone, dirt and coal. (I used 2 Ramps next to each other to have an easier way down ;)

Recreation: Pull a wooden cart down the ramps and have some fun while waiting to get down again ;)

**Expected behavior:**
A player can smoothly go down a block with a cart aswell as up.

**Actual behavior:**
The slope would catapult the player with the cart into the sky. The forward velocity did not change, however, upward momentum was gained.

**Comment by SeanAtr0n** [comment]
This happens for you anywhere in the world or in specific locations?

**Comment by PacifyerGrey** [comment]
Not the exact situation but I got several space flights with a cart too.
Most of mine were caused by trying to squeeze a cart between two objects like tree seeblings. The cart most probably begins to cyclicly bounce between the two with increasingly growing momentum and at some point is set free flying a great distance (not only vertical).

Same behaviour was observed with rubble piles when they start to shiver and shoot some rubble pieces with a great velocity causing those pieces to end up on the surface above the mine.

And trees if growing close to each other (like three blocks grid) when chopped down will likely shoot chopped tree to the side in many cases, or make it spin like mad.

So the source of the problem is more generic and is caused by some weird math which increases object momentum on impact while it most likely should decrease it.
